src/libs/histogram.c

```diff
@@ -266,6 +275,39 @@ static void _lib_histogram_process_waveform(dt_lib_histogram_t *d, const float *
   }
 }
 
+#ifdef _OPENMP
+#pragma omp declare simd aligned(rgb, Luv:16)
+#endif
+static inline void rgb_to_chroma(const float rgb[4],
+                                 float Luv[4],
+                                 dt_iop_order_iccprofile_info_t *prof,
+                                 dt_lib_histogram_vectorscope_type_t cs)
+{
+  float XYZ_D50[4] DT_ALIGNED_PIXEL;
+  // NOTE: see for comparison/reference rgb_to_JzCzhz() in color_picker.c
+  // FIXME: is there a way to optimize this chain of conversions?
+  // this goes to the PCS which has standard illuminant D50
+  dt_ioppr_rgb_matrix_to_xyz(rgb, XYZ_D50, prof->matrix_in, prof->lut_in,
+                             prof->unbounded_coeffs_in, prof->lutsize,
+                             prof->nonlinearlut);
+  if(cs == DT_LIB_HISTOGRAM_VECTORSCOPE_CIELUV)
+  {
+    // FIXME: do have to worry about chromatic adaptation?
+    float xyY_D50[4] DT_ALIGNED_PIXEL;
+    dt_XYZ_to_xyY(XYZ_D50, xyY_D50);
+    // FIXME: this is u*v* (D50 corrected), more helpful than u'v', yes?
```

Comment:
yes

Reply:
This is "yes" to the 1st ("chromatic adaptation") or the 2nd, that u*v* is more helpful?

RE 1st: I _think_ we're OK: the `u*v*` transform in `dt_xyY_to_Luv()` assumes a D50 whitepoint, which I think is OK, as the `RGB` -> `xyY` transform is adapted for D50.

RE 2nd: I think we want `u*v*` (and D50 corrected is fine -- as we're plotting relative to whitepoint) for a vectorscope. For a full "chromaticity diagram" plot -- not in this PR, we'd want `u'v'` (or `xy`, etc.).

Please let me know if any of this reasoning is faulty.

Regarding point 1, above: I know an alternative could be to the transform to PCS without any chromatic adaptation (would take some work, but possible), then make the transformation to `u*v*` be based on the histogram profile whitepoint. I can test this, but I'm not sure if this'll produce much of a different result.

I've been working for days to wrap my head around chromatic adaptation in the case of a chromaticity diagram (`xy` or `u'v'` plot, showing the whole spectral locus). As far as I can think it through, the `RGB` -> `XYZ` transform really should be absolute colorimetric. Otherwise the whitepoint will adapt to being D50 (the PCS whitepoint), and the plot will always appear to have a D50 whitepoint and the primaries won't match their known values. Unfortunately, darktable doesn't currently have code to generate an absolute colorimetric matrix. And LCMS pre-adapts the internal profiles (sRGB, Adobe RGB, Rec709, etc.) built in colorspaces.c to D50, so there's no easy way whether to un-adapt them to make a absolute colorimetric intent matrix.

I've got a hacky solution in another branch. Essentially, I hardcode that ProPhoto is the one internal profile which is D50, and for all the rest, the route is `RGB` -> `XYZ` (D50) then `dt_XYZ_D50_2_XYZ_D65()` to un-adapt. For profiles loaded from a file, there appears to be a meaningful whitepoint tag, which makes things easier.

A proper solution may be to record whitepoint of internal profiles on their creation. And then make a hacked "unadapted" matrix for the conversion. A really proper solution would be to make `dt_ioppr_generate_profile_info()` or such do this work.

Reply:
No. The chromatic adaptation is not needed for CIE Luv since we output any RGB with D50-massaged primaries, so white point is D50 anytime. When saving to raster output, we actuall save a Bradford transform in the `chad` tag, so the primaries are massaged back to D65 by whatever software will open them. For JzAzBz, however, we need to transform to D65, and that can be achieved by a matrix product with : 

```
  [[  9.80760485e-01,  -4.25541784e-17,  -7.61959005e-19],
    [  4.82934624e-17,   1.01555271e+00,  -7.63213113e-18],
    [ -6.47162968e-19,  -5.69389701e-19,   1.30191586e+00]]
  * ```

Yes, Lu*v* is better than Lu'v' because it is more evenly spaced.

Reply:
> For JzAzBz, however, we need to transform to D65, and that can be achieved by a matrix product

This is a CAT16 transform matrix, yes? The code currently uses Bradford adaptation matrix via `dt_XYZ_D50_2_XYZ_D65()`. It doesn't appear to be a huge difference (< 1% in results, at a quick test).

Question: All the built-in color profiles except ProPhoto are adapted to D50 with a Bradford transform. Even if CAT16 transform is "better", wouldn't "un-adapting" them back to D65 via Bradford produce the closest to pre-adaptation values?

The adaptation appears to be happening in LCMS:

https://github.com/darktable-org/darktable/blob/88067a4e808891306a8db35e909dabb5c57fa084/src/common/colorspaces.c#L285-L295

This goes to:

https://github.com/mm2/Little-CMS/blob/8154d6968674d3a6a4ab05ae85c4d27de1b97a7c/src/cmsvirt.c#L143-L152

then to:

https://github.com/mm2/Little-CMS/blob/8154d6968674d3a6a4ab05ae85c4d27de1b97a7c/src/cmswtpnt.c#L231-L245

Reply:
I finally tested this by converting primaries in RGB via the profile (`dt_ioppr_rgb_matrix_to_xyz()`) to D50 XYZ, converting them to D65, and comparing them with the "official" primaries.

For most cases, the code in this PR, which converts XYZ D50 -> XYZ D65 via `dt_XYZ_D50_2_XYZ_D65()` is right. For internal profiles, this reverses the Bradford adaptation used to adapt primaries to D50 on profile construction. I think this is an ICC standard (at least for V2?) and seems OK for profiles from files.

As an example, Adobe RGB primaries

```
correct (CIE xy)
Red 	0.6400 	 	0.3300
Green 	0.2100 	 	0.7100
Blue 	0.1500 	 	0.0600

-> D65 via Bradford
Red 	0.640017 	0.330004
Green 	0.210027 	0.709998
Blue 	0.150010 	0.060011

-> D65 via CAT16
Red 	0.636652 	0.336368
Green 	0.219789 	0.693687
Blue 	0.124011 	0.054409
```

Where this falls down is if the profile is already in D50 (ProPhoto). Then presumably the CAT16 adaptation will be slightly "better"? It'd be possible to make the code smart enough to recognize that the profile is D50 and do that.